src/librustc_ast/ast.rs

```diff
@@ -309,19 +310,48 @@ pub type GenericBounds = Vec<GenericBound>;
 /// Specifies the enforced ordering for generic parameters. In the future,
 /// if we wanted to relax this order, we could override `PartialEq` and
 /// `PartialOrd`, to allow the kinds to be unordered.
-#[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
+#[derive(PartialEq, Eq, Hash, Clone, Copy)]
 pub enum ParamKindOrd {
     Lifetime,
     Type,
-    Const,
+    // unordered is only true if sess.has_features().const_generics
+    // is active. Specifically, if it's only `min_const_generics`, it will still require ordered
+    // types.
```

Comment:
```suggestion
    // `unordered` is only `true` if `sess.has_features().const_generics`
    // is active. Specifically, if it's only `min_const_generics`, it will still require ordered
    // types.
```
nit: also what are "ordered types" here

Reply:
ah yea I can rephrase that to make sense my bad